gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java

```diff
@@ -116,12 +126,15 @@ public HttpJsonCallContext merge(ApiCallContext inputCallContext) {
       newCredentials = this.credentials;
     }
 
-    return new HttpJsonCallContext(newChannel, newDeadline, newCredentials);
+    ImmutableMap<String, List<String>> newExtraHeaders =
+        mergeExtraHeaders(this.extraHeaders, httpJsonCallContext.extraHeaders);
```

Comment:
It is not a strict rule, but usually `this.` is used only if it is required (otherwise it would be a name conflict). This happens usually either in constructor, field setter or field getter method. In other places `this` is omitted if possible. The idea is to not make language more verbose than needed (java is already pretty verbose). Pleaser remove explicit `this.`, if not required (similar thing on line 137, 193, 200, 222, 226; note `this` on line 206 is ok, since it is a getter).